### DISCUSSION TITLE
Looking for an example using TabView with dynamic items

### DISCUSSION THREAD

**Discussion by balintn22** [xamarin/XamarinCommunityToolkit#1800] [Q&A]
I need to implement a Xamarin page with a tab view, with tab pages dymanically created.
I tried using TabPage, but I need extra content 8vaigation) at the bottom of the screen.
I think TabView is the control to use.
However, all the example I can find use a single .xaml page defining the TabView and static pages within.
Is there an example that would demonstrate how the tab pages / tab page template can be defined and how they work with the parent TabView?

**Comment by balintn22** [comment]
Environment: Xamarin 5, Shell application
I tried to use the example provided with the Xamarin Comunity Toolkit soureces [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/samples/XCT.Sample/Pages/Views/TabView/CustomTabsPage.xaml), but I couldn't make it work, the pages never rendered.
The way I tried was by setting the TabView ItemsSource property, and declaring TabContentDataTemplate and TabViewItemDataTemplate as page level resources. Displaying static tab pages was easy, I could not make the dynamic work.
I am using dependency injection with a locator, maybe this had something to do with it.

I ended up using TabView, add pages from code behind (not using .xaml for the pages).

**Comment by sjordanGSS** [comment]
Hi, I've managed to do what you've discussed here in an app in the past, it's closed source so no example code sadly, but you can bind the ItemsSource to a list in your viewmodel, then use TabContentDataTemplate to display properties on your models within the tabs. This example here shows you how to do it: https://github.com/xamarin/XamarinCommunityToolkit/blob/main/samples/XCT.Sample/Pages/Views/TabView/TabItemsSourcePage.xaml
